diff --git a/src/sugar4/graphics/icon.py b/src/sugar4/graphics/icon.py
index e7d0b08a..eb4a91f4 100644
--- a/src/sugar4/graphics/icon.py
+++ b/src/sugar4/graphics/icon.py
@@ -249,7 +249,7 @@ class _IconBuffer:
                 self.width,
                 1,
                 Gtk.TextDirection.NONE,
-                Gtk.IconLookupFlags.NONE,
+                0,
             )
 
             if icon_paintable:
@@ -300,7 +300,7 @@ class _IconBuffer:
             size,
             1,
             Gtk.TextDirection.NONE,
-            Gtk.IconLookupFlags.NONE,
+            0,
         )
 
         if not badge_paintable:
@@ -1055,7 +1055,7 @@ def get_icon_file_name(icon_name: str) -> Optional[str]:
         STANDARD_ICON_SIZE,
         1,
         Gtk.TextDirection.NONE,
-        Gtk.IconLookupFlags.NONE,
+        0,
     )
 
     if icon_paintable:
diff --git a/src/sugar4/graphics/iconentry.py b/src/sugar4/graphics/iconentry.py
index ab104aad..f5645d3a 100644
--- a/src/sugar4/graphics/iconentry.py
+++ b/src/sugar4/graphics/iconentry.py
@@ -99,7 +99,7 @@ class IconEntry(Gtk.Entry):
             _ICON_SIZE,
             1,
             Gtk.TextDirection.NONE,
-            Gtk.IconLookupFlags.NONE,
+            0,
         )
 
         if not icon_paintable:
