--- a/src/jarabe/view/service.py
+++ b/src/jarabe/view/service.py
@@ -17,6 +17,7 @@
 """D-bus service providing access to the shell's functionality"""
 
 import dbus
+import os
 from dbus import service
 
 from jarabe.model import shell
@@ -91,8 +92,14 @@
         activity = self._shell_model.get_activity_by_id(activity_id)
         if activity is None:
             return False
+        pid = activity.get_pid()
         activity.stop()
         activity.close_window()
+        if pid:
+            try:
+                os.kill(pid, 15)
+            except ProcessLookupError:
+                pass
         return True
     @service.method(_DBUS_SHELL_IFACE, in_signature='', out_signature='b')
     def ShowJournal(self):
